### PR TITLE
Pass consistency guarantees to the dispatcher

### LINKED ISSUE #2868: TransactionSettings isn't a good abstraction
As of #2828 we pass transaction info to each message pump instance. This allows satellites and the main pipe to have different TX settings. All good so far. 

Problem is that the `TransactionSettings` class is a mess. We need to clean things up.

### PR DESCRIPTION
**Pull request by andreasohlund**
This will allow the outbox to request the message to not be enlisted in any receive transactions

* Introduced `ExactlyOnce`and `AtMostOnce` (will be refactored to `enum` in a separate pull)
* Satellites are now required to specify what level of consistency they want
* Think we have an issue with the TM satellites #2885 
* `AtomicWithReceive` is now a `DispatchConsistency.Isolated`

Fixes #2868

### CODE REVIEW

**Comment by andreasohlund** [comment]
@Particular/tf-asyncawait and @SzymonPobiega please review

**Comment by andreasohlund** [comment]
@Particular/tf-asyncawait please review

**Comment by andreasohlund** [comment]
@timbussmann @weralabaj please review since you'll need this one to fix the TM bug on v6

File: src/NServiceBus.AcceptanceTests/Tx/When_receiving_with_dtc_disabled.cs [hunk 1]
```diff
@@ -43,9 +43,7 @@ public class NonDTCEndpoint : EndpointConfigurationBuilder
         {
             public NonDTCEndpoint()
             {
-                EndpointSetup<DefaultServer>(c => c.Transactions()
-                    .DisableDistributedTransactions()
-                    .WrapHandlersExecutionInATransactionScope());
+                EndpointSetup<DefaultServer>(c => c.Transactions().DisableDistributedTransactions().WrapHandlersExecutionInATransactionScope());
```
Comment (SzymonPobiega) [inline src/NServiceBus.AcceptanceTests/Tx/When_receiving_with_dtc_disabled.cs]: nitpicking: why single line? I need to scroll now :\

Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Tx/When_receiving_with_dtc_disabled.cs]: Agreed, will fix

File: src/NServiceBus.Core/Transports/Msmq/MessagePump.cs [hunk 2]
```diff
@@ -41,7 +41,7 @@ public void Init(Func<PushContext, Task> pipe, PushSettings settings)
             inputQueue = new MessageQueue(inputAddress.FullPath, false, true, QueueAccessMode.Receive);
             errorQueue = new MessageQueue(errorAddress.FullPath, false, true, QueueAccessMode.Send);
 
-            if (settings.TransactionSettings.IsTransactional && !QueueIsTransactional())
+            if (settings.RequiredConsistency.GetType() != typeof(AtMostOnce) && !QueueIsTransactional())
```
Comment (SzymonPobiega) [inline src/NServiceBus.Core/Transports/Msmq/MessagePump.cs]: I am not a big fan of using types in `if` statements but I accept this an an intermediate step. I believe the end goal might be to have a pump implementation per consistency mode. For now :ok:

**Comment by SzymonPobiega** [comment]
Good to go. Shall I merge it @andreasohlund ?

File: src/NServiceBus.Core/Transports/Msmq/MessagePump.cs (hunk 2 above)
Comment (andreasohlund) [inline src/NServiceBus.Core/Transports/Msmq/MessagePump.cs]: Agreed, I'm about to make the consistency guarantees a `enum` to fix this

**Pull request merged**